js/src/ui/TxList/store.js

```diff
@@ -14,150 +14,121 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-import { action, observable, transaction } from 'mobx';
-import { uniq } from 'lodash';
+import { action, observable } from 'mobx';
 
 export default class Store {
   @observable blocks = {};
   @observable sortedHashes = [];
   @observable transactions = {};
 
-  constructor (api) {
+  constructor (api, newError) {
     this._api = api;
-    this._subscriptionId = 0;
-    this._pendingHashes = [];
-
-    this.subscribe();
+    this._newError = newError;
   }
 
-  @action addBlocks = (blocks) => {
-    this.blocks = Object.assign({}, this.blocks, blocks);
+  @action addHash = (hash) => {
+    if (!this.sortedHashes.includes(hash)) {
+      this.sortedHashes.push(hash);
+      this.sortHashes();
```

Comment:
Arrays are fast, but instead of sorting in `O(n log(n))` each time you instert an element you can just insert with `O(log(n))` complexity and for large number of elements this will matter.

But nevermind, now I see this is just used to store local transactions so this will be 25 transactions at most afair so it's not even worth thinking about :)